libdevice/include/sanitizer_device_utils.hpp

```diff
@@ -39,5 +40,6 @@ class
 private:
   T val;
 };
+#endif
 
-enum DeviceType : uintptr_t { UNKNOWN, CPU, GPU_PVC, GPU_DG2 };
+enum DeviceType : uint64_t { UNKNOWN, CPU, GPU_PVC, GPU_DG2 };
```

Comment:
should uint32_t be enough here?

Reply:
I tried uint32_t, but level zero failed to write global variable in uint32_t.